langchain/tools/base.py

```diff
@@ -78,36 +92,43 @@ def run(
 
     async def arun(
         self,
-        tool_input: str,
+        tool_input: Union[str, Dict],
         verbose: Optional[bool] = None,
         start_color: Optional[str] = "green",
         color: Optional[str] = "green",
         **kwargs: Any
     ) -> str:
         """Run the tool asynchronously."""
+        if isinstance(tool_input, str):
+            if len(self.args) > 1:
+                raise ValueError("Cannot call run on tools with > 1 argument")
```

Comment:
I think you might need to update this error message, something like "expected args of type dict, received of type string"